### PR TITLE
Optimizations and UX improvements to the dashboard screen

### PR DESCRIPTION
**Pull request by davidfischer**
- No longer show how many builds a project has. This is expensive and not very useful
- Instead, show the date of the last build
- Fixes a performance issue affecting the corporate site

Note that I used the [HTML5 `<time>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time) to represent the datetime. This tag has attributes that have a machine readable datetime as well as a title with a precise locale-specific human-readable datetime as opposed to the nebulous "a year ago". I think we should use this convention or something like it more widely.

This PR builds on some optimizations in #5472 and this should be refactored slightly if that is merged first.

## Screenshot
Screen Shot 2019-04-26 at 8 58 09 PM

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -845,6 +845,12 @@ def get_latest_build(self, finished=True):
 
         :param finished: Return only builds that are in a finished state
         """
+        if hasattr(self, '_latest_build'):
+            # Cached latest build
```
[the remark this answers is not in the document]
Comment (davidfischer) [reply readthedocs/projects/models.py]: I agree. How about after https://github.com/rtfd/readthedocs.org/pull/5472 is merged I'll refactor this slightly to have a class variable for caching it.

File: readthedocs/projects/querysets.py [hunk 2]
```diff
@@ -80,7 +80,22 @@ def is_active(self, project):
...
     def dashboard(self, user=None):
-        return self.for_admin_user(user)
+        """Get the projects for this user including the latest build"""
+        from readthedocs.builds.models import Build
+
+        projects = self.for_admin_user(user)
+
+        # Prefetch the latest build for each project.
+        subquery = Subquery(
+            Build.objects.filter(project=OuterRef('project_id')).values_list('id', flat=True)[:1]
```
[the remark this answers is not in the document]
Comment (davidfischer) [reply readthedocs/projects/querysets.py]: It isn't necessary due to the default ordering but maybe it is best to be explicit.

**Comment by davidfischer** [comment]
I refactored this optimization to be on the manager instead of sprinkled in a few functions.

**Comment by saadmk11** [comment]
Seems like a better approach :)

**Pull request merged**